From 65d805fdfc5ceead2645d3107cbae7b7696a1f15 Mon Sep 17 00:00:00 2001
From: Willy Tarreau <w@1wt.eu>
Date: Tue, 15 Jul 2014 18:05:58 +0200
Subject: [PATCH 43/53] BUILD: fix dependencies between config and compat.h

compat.h only depends on the system, and config needs compat, not the
opposite. global.h was fixed to explicitly include standard.h for LONGBITS.
---
 include/common/compat.h | 2 --
 include/common/config.h | 1 +
 include/types/global.h  | 1 +
 3 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/include/common/compat.h b/include/common/compat.h
index 48ea1f7..ecbc3b1 100644
--- a/include/common/compat.h
+++ b/include/common/compat.h
@@ -27,8 +27,6 @@
 #include <sys/types.h>
 #include <sys/socket.h>
 #include <arpa/inet.h>
-#include <common/config.h>
-#include <common/standard.h>
 
 #ifndef BITS_PER_INT
 #define BITS_PER_INT    (8*sizeof(int))
diff --git a/include/common/config.h b/include/common/config.h
index 5833cfc..27b8f14 100644
--- a/include/common/config.h
+++ b/include/common/config.h
@@ -23,6 +23,7 @@
 #define _COMMON_CONFIG_H
 
 #include <common/compiler.h>
+#include <common/compat.h>
 #include <common/defaults.h>
 
 /* this reduces the number of calls to select() by choosing appropriate
diff --git a/include/types/global.h b/include/types/global.h
index 14a6bf2..85203fa 100644
--- a/include/types/global.h
+++ b/include/types/global.h
@@ -25,6 +25,7 @@
 #include <netinet/in.h>
 
 #include <common/config.h>
+#include <common/standard.h>
 #include <types/freq_ctr.h>
 #include <types/listener.h>
 #include <types/proxy.h>
-- 
1.8.5.5

